dlt/destinations/synapse/sql_client.py

```diff
@@ -0,0 +1,164 @@
+            self.commit_transaction()
+        except Exception:
+            self.rollback_transaction()
+            raise
+
+    @raise_database_error
+    def commit_transaction(self) -> None:
+        self._conn.commit()
+        self._conn.autocommit = True
+
+    @raise_database_error
+    def rollback_transaction(self) -> None:
+        self._conn.rollback()
+        self._conn.autocommit = True
+
+    @property
+    def native_connection(self) -> pyodbc.Connection:
+        return self._conn
+
+    def drop_dataset(self) -> None:
+        # MS Sql doesn't support DROP ... CASCADE, drop tables in the schema first
+        # Drop all views
+        rows = self.execute_sql(
+            "SELECT table_name FROM information_schema.views WHERE table_schema = %s;", self.dataset_name
+        )
+        view_names = [row[0] for row in rows]
+        self._drop_views(*view_names)
+        # Drop all tables
+        rows = self.execute_sql(
+            "SELECT table_name FROM information_schema.tables WHERE table_schema = %s;", self.dataset_name
+        )
+        table_names = [row[0] for row in rows]
+        self.drop_tables(*table_names)
+
+        self.execute_sql("DROP SCHEMA %s;" % self.fully_qualified_dataset_name())
+
+    def drop_tables(self, *tables: str) -> None:
+        if not tables:
+            return
+        statements = [f"DROP TABLE {self.make_qualified_table_name(table)};" for table in tables]
+        self.execute_fragments(statements)    
+    
+    def _drop_views(self, *tables: str) -> None:
+        if not tables:
+            return
+        statements = [f"DROP VIEW {self.make_qualified_table_name(table)};" for table in tables]
+        self.execute_fragments(statements)
+
+    def execute_sql(self, sql: AnyStr, *args: Any, **kwargs: Any) -> Optional[Sequence[Sequence[Any]]]:
+        with self.execute_query(sql, *args, **kwargs) as curr:
+            if curr.description is None:
+                return None
+            else:
+                f = curr.fetchall()
+                return f
+
+    @contextmanager
+    @raise_database_error
+    def execute_query(self, query: AnyStr, *args: Any, **kwargs: Any) -> Iterator[DBApiCursor]:
+        assert isinstance(query, str)
+        curr: DBApiCursor = None
+        if kwargs:
+            raise NotImplementedError("pyodbc does not support named parameters in queries")
+        if args:
+            # TODO: this is bad. See duckdb & athena also
+            query = query.replace("%s", "?")
+        curr = self._conn.cursor()
+        try:
+            print("Executing query:")
+            print(query)
+            # unpack because empty tuple gets interpreted as a single argument
+            # https://github.com/mkleehammer/pyodbc/wiki/Features-beyond-the-DB-API#passing-parameters
+            curr.execute(query, *args)
+            yield DBApiCursorImpl(curr)  # type: ignore[abstract]
+        except pyodbc.Error as outer:
+            print(f"Error in query:")
```

Comment:
debug code

Reply:
This and all other debug code will be removed once the rest of the PR is set / tests are resolved